src/densities.jl

```diff
@@ -24,42 +25,42 @@ grid `basis`, where the individual k-points are occupied according to `occupatio
     ρ = similar(ψ[1], T, (basis.fft_size..., basis.model.n_spin_components))
     ρ .= 0
     ψnk_real = zeros(complex(T), basis.fft_size)
-    for ik = 1:length(basis.kpoints)
-        kpt = basis.kpoints[ik]
+    for (ik, (weight, kpt)) in enumerate(zip(basis.kweights, basis.kpoints))
```

Comment:
not too sure about that, enumerate(zip) is really the place where I feel it's simpler to just iterate over indices...